pulsar-broker/src/test/java/org/apache/pulsar/broker/service/schema/SchemaServiceTest.java

```diff
@@ -154,6 +168,18 @@ public void writeReadBackDeleteSchemaEntry() throws Exception {
         assertNull(schemaRegistryService.getSchema(schemaId1).get());
     }
 
+    @Test
+    public void testConcurrentPutSchema() throws Exception {
+        String schemaId = "tenant/ns/topic_test_concurrent_put_schema" + UUID.randomUUID();
+        CompletableFuture.allOf(
+                putSchema(schemaId, schemaData1),
+                putSchema(schemaId, schemaData1),
+                putSchema(schemaId, schemaData1),
+                putSchema(schemaId, schemaData1))
+                .get();
+        assertEquals(getAllSchemas(schemaId).size(), 1);
```

Comment:
IIUC,  the first put schema request will success ,and the second request will fail with `BadVersionException`. Then the second request will retry call `putSchemaIfAbsent`, result with `schema is already exists`.  That's OK.

But I have a minor question, the second request will also create some ledgers, which will never be used again, will this PR try to delete the unused ledgers？

Reply:
OK, how about changing the title of this PR to `fix creating lots of versions for the same schema`. Because it seems that this PR doesn't fix creating lots of ledgers,  instead it fix creating lot of versions. @aloyszhang

Reply:
Sounds good.
One point to clarify is "creating lot of versions" will create lots of new schema ledger.